tools/rccl-prim-test/rccl_prim_test.cpp

```diff
@@ -122,21 +142,83 @@ do {                                                                           \
   }                                                                            \
 } while (0)
 
-static void setupPeers() {
-    int deviceCnt, dev;
-
-     HIPCHECK(hipGetDeviceCount(&deviceCnt));
-     HIPCHECK(hipGetDevice(&dev));
-    //! If gpus are not peer enabled, enable them
-    for (int i = 0; i < deviceCnt; i++) {
-         HIPCHECK(hipSetDevice(i));
-        for (int j = 0; j < deviceCnt; j++) {
-            if (i != j) {
-                HIPCHECK(hipDeviceEnablePeerAccess(j, 0));
-            }
+static void setupPeers(uint32_t *info) {
+  int deviceCnt, dev;
+
+  HIPCHECK(hipGetDeviceCount(&deviceCnt));
+  HIPCHECK(hipGetDevice(&dev));
+  //! If gpus are not peer enabled, enable them
+  for (int i = 0; i < deviceCnt; i++) {
+    HIPCHECK(hipSetDevice(i));
+    for (int j = 0; j < deviceCnt; j++) {
+      if (i != j) {
+	int p2p;
+        HIPCHECK(hipDeviceCanAccessPeer(&p2p, i, j));
+        if (!p2p) {
+          printf("Cannot enable peer access between device %d and %d. You may use HIP_VISIBLE_DEVICES to limit GPUs.\n",
+           i, j);
+          exit(-1);
         }
+        HIPCHECK(hipDeviceEnablePeerAccess(j, 0));
+        uint32_t linktype;
+        HIPCHECK(hipExtGetLinkTypeAndHopCount(i, j, &linktype, &info[i*deviceCnt+j]));
+      }
+      else
+        info[i*deviceCnt+j] = 0;
+    }
+  }
+  HIPCHECK(hipSetDevice(dev));
+}
+
+static void printRing(int id, int *ring, int deviceCnt) {
```

Comment:
Can we pass in info so that we can also see the hop distance in between?

Reply:
You can always refer to distance which are already printed in connection matrix. Ring is to show the constructed GPU order.
Connection matrix:
 0  1  1  2 
 1  0  2  1 
 1  2  0  1 
 2  1  1  0 
Ring 0: 0 1 3 2 
Ring 1: 0 2 3 1